WordPress/Classes/ViewRelated/Site Creation/DesignSelection/SiteDesignContentCollectionViewController.swift

```diff
@@ -120,10 +122,12 @@ class SiteDesignContentCollectionViewController: CollapsableHeaderViewController
             return
         }
         let design = siteDesigns[selectedIndexPath.row]
+        SiteCreationAnalyticsHelper.trackSiteDesignSelected(design)
```

Comment:
Not sure if the key could ever not be present, but should we have a guard here?

Reply:
> Not sure if the key could ever not be present

I'm not 100% sure which key you mean based on where the comment is appearing in GH. So let me try to address both lines and see what you think :) 

If you're talking about the `selectedIndexPath.row` here for the array then we are unwrapping that here on the lines above.

https://github.com/wordpress-mobile/WordPress-iOS/blob/fd8ee82c295a792cd705dd9b7ac6d7346f99c2eb/WordPress/Classes/ViewRelated/Site%20Creation/DesignSelection/SiteDesignContentCollectionViewController.swift#L120-L123

I thought about adding an additional error logging but it would have been a wordsmithed version of: "I have no idea how this could happen but if it does here's a log" 😄 If we hit that scenario we treat it like a skip. If we wanted to track who falls under that flow we would probably still be able to do that since selecting a design and skipping both have an analytics call. So the lack of either of those events could be a separate funnel.  

---

For `design` if the `selectedIndexPath` path is beyond the bounds of the array (which would cause a crash) that would have to mean that a successful network request occurred, and then happened again and received different data. In the current state of the code, the API results are only fetched once and held in local memory until the view is destroyed. So there isn't really a code path that would allow this. `#famousLastWords` lol

Reply:
> "I have no idea how this could happen but if it does here's a log"

:smile: 

My apologies .. I selected the wrong line :man_facepalming: . I meant the `design`, I was wondering if the row was not present, if `trackSiteDesignSelected` would choke on a `nil`. But your explanation makes sense, thanks for clarifying! LGTM!